src/Log.test.ts

```diff
@@ -11,18 +12,28 @@ import { getMock, getMockCallArg } from "./testHelpers";
 import assert from "node:assert";
 
 describe("Log", () => {
+  class SpyAdaptor extends LogAdaptor {
+    append = test.mock.fn();
+  }
+  let log: Log;
+  let adaptor: LogAdaptor;
+  beforeEach(() => {
+    log = new Log();
+    adaptor = new SpyAdaptor();
+    log.addAdaptor(adaptor);
+  });
+
   describe("addSubject", () => {
-    class SpyAdaptor extends LogAdaptor {
-      append = test.mock.fn();
-    }
-    const adaptor = new SpyAdaptor();
-    const bananas = new LogSubject("bananas");
+    const bananasData = {};
```

Comment:
Added a log feature: Subjects have an optional data parameter. If specified, this data becomes the identity of the subject. Otherwise the subject's object itself is the identity. This is intended to be used with filtering to facilitate reading log messages about particular objects (e.g. live entities.)